src/main/java/io/mbab/sda/groupproject/repository/CountryRepository.java

```diff
@@ -4,28 +4,39 @@
 import lombok.RequiredArgsConstructor;
 
 import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
 import javax.persistence.criteria.CriteriaBuilder;
 import java.util.List;
+import java.util.Optional;
 
 @RequiredArgsConstructor
 public class CountryRepository implements CrudRepository<Country, Integer> {
 
   private final EntityManager em;
-  //Class <Country> entityClass;
-
-  public Country findByName(String name){
-    CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
-    var criteriaQuery = criteriaBuilder.createQuery(Country.class);
-    var root = criteriaQuery.from(Country.class);
-    return em.createQuery(
-                    criteriaQuery.select(root).where(criteriaBuilder.equal(root.get("name"), name)))
-                    .getSingleResult();
-
+  // Class <Country> entityClass;
+
+  public Optional<Country> findByName(String name) {
+    Optional<Country> country = null;
+    try {
+      CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+      var criteriaQuery = criteriaBuilder.createQuery(Country.class);
+      var root = criteriaQuery.from(Country.class);
+      country =
```

Comment:
var country = em.createQuery(
                      criteriaQuery
                          .select(root)
                          .where(criteriaBuilder.equal(root.get("name"), name)))
                  .getSingleResult();

return Optional.of(country);